File: internal/dto/product_dto.go
package dto

type CreateProductRequest struct {
	Name     string  `json:"name" binding:"required,min=2,max=150"`
	SKU      *string `json:"sku"`
	LengthMM float64 `json:"length_mm" binding:"required,gt=0"`
	WidthMM  float64 `json:"width_mm" binding:"required,gt=0"`
	HeightMM float64 `json:"height_mm" binding:"required,gt=0"`
	WeightKG float64 `json:"weight_kg" binding:"required,gt=0"`
	ColorHex *string `json:"color_hex" binding:"omitempty,hexcolor"`
}

type UpdateProductRequest struct {
	Name     string  `json:"name" binding:"required,min=2,max=150"`
	SKU      *string `json:"sku"`
	LengthMM float64 `json:"length_mm" binding:"required,gt=0"`
	WidthMM  float64 `json:"width_mm" binding:"required,gt=0"`
	HeightMM float64 `json:"height_mm" binding:"required,gt=0"`
	WeightKG float64 `json:"weight_kg" binding:"required,gt=0"`
	ColorHex *string `json:"color_hex" binding:"omitempty,hexcolor"`
}

type ProductResponse struct {
	ID       string  `json:"id"`
	Name     string  `json:"name"`
	SKU      *string `json:"sku,omitempty"`
	LengthMM float64 `json:"length_mm"`
	WidthMM  float64 `json:"width_mm"`
	HeightMM float64 `json:"height_mm"`
	WeightKG float64 `json:"weight_kg"`
	ColorHex *string `json:"color_hex,omitempty"`
}

File: mobile/lib/dtos/product_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'product_dto.g.dart';

@JsonSerializable()
class ProductResponseDto {
  final String id;
  final String name;
  @JsonKey(name: 'length_mm')
  final double lengthMm;
  @JsonKey(name: 'width_mm')
  final double widthMm;
  @JsonKey(name: 'height_mm')
  final double heightMm;
  @JsonKey(name: 'weight_kg')
  final double weightKg;
  @JsonKey(name: 'color_hex')
  final String? colorHex;

  ProductResponseDto({
    required this.id,
    required this.name,
    required this.lengthMm,
    required this.widthMm,
    required this.heightMm,
    required this.weightKg,
    this.colorHex,
  });

  factory ProductResponseDto.fromJson(Map<String, dynamic> json) => _$ProductResponseDtoFromJson(json);
}

@JsonSerializable()
class CreateProductRequestDto {
  final String name;
  @JsonKey(name: 'length_mm')
  final double lengthMm;
  @JsonKey(name: 'width_mm')
  final double widthMm;
  @JsonKey(name: 'height_mm')
  final double heightMm;
  @JsonKey(name: 'weight_kg')
  final double weightKg;
  @JsonKey(name: 'color_hex')
  final String? colorHex;

  CreateProductRequestDto({
    required this.name,
    required this.lengthMm,
    required this.widthMm,
    required this.heightMm,
    required this.weightKg,
    this.colorHex,
  });

  Map<String, dynamic> toJson() => _$CreateProductRequestDtoToJson(this);
}

@JsonSerializable()
class UpdateProductRequestDto {
  final String name;
  @JsonKey(name: 'length_mm')
  final double lengthMm;
  @JsonKey(name: 'width_mm')
  final double widthMm;
  @JsonKey(name: 'height_mm')
  final double heightMm;
  @JsonKey(name: 'weight_kg')
  final double weightKg;
  @JsonKey(name: 'color_hex')
  final String? colorHex;

  UpdateProductRequestDto({
    required this.name,
    required this.lengthMm,
    required this.widthMm,
    required this.heightMm,
    required this.weightKg,
    this.colorHex,
  });

  Map<String, dynamic> toJson() => _$UpdateProductRequestDtoToJson(this);
}
